Lib/typing.py

```diff
@@ -253,8 +253,8 @@ def _collect_parameters(args):
         if hasattr(t, '__typing_subst__'):
             if t not in parameters:
                 parameters.append(t)
-        # We need to avoid bare Python types as those aren't really objects.
-        elif type(t) is not type:
+        # Ensures we aren't grabbing from something non-generic. E.g. a bare Python class isn't generic.
+        elif isinstance(t, (_GenericAlias, GenericAlias, types.UnionType)):
```

Comment:
Curious what @serhiy-storchaka would think of this reversal. And are there tests that cover all cases? It might be more elegant if there was a common base class, but that would have to be a C class, so never mind.